### PR TITLE
Add semantic logger formatter for data dog

### PR DESCRIPTION
**Pull request by cwrw**
this is taken from:
https://github.com/flipgroup/data-dog-semantic-logger/

The amendments:
1) log does not have method `method_name`, I adapted it to the data dog
recommendations (added custom name for logger name and the log.nam was
actually the class name)
2) Rubocop fixes
3) Remove try so that we can see the errors for anything missing

### CODE REVIEW

File: app/lib/semantic_logger/formatters/data_dog.rb [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+      def time?
+        !time.nil?
+      end
+
+      def appname
+        ENV['APP_NAME'].to_s
+      end
+
+      def appname?
+        !appname.nil?
```
Comment (tavy88) [inline app/lib/semantic_logger/formatters/data_dog.rb]: Can this not be `appname.present?` instead?

Comment (cwrw) [reply app/lib/semantic_logger/formatters/data_dog.rb]: not nil is not synonymous to present, example:
```
irb(main):002:0> !{}.nil?
=> true
irb(main):003:0> {}.present?
=> false
```

Comment (tavy88) [reply app/lib/semantic_logger/formatters/data_dog.rb]: Surely your `appname` can be NIL or have value though. Why would it have a hash value?

```ruby
irb(main):001:0> !nil.nil?
=> false
irb(main):002:0> nil.present?
=> false
```

File: app/lib/semantic_logger/formatters/data_dog.rb [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+        (local_time.to_time.utc.to_f * 1000).to_i
+      end
+
+      def time?
+        !time.nil?
+      end
+
+      def appname
+        ENV['APP_NAME'].to_s
```
Comment (tavy88) [inline app/lib/semantic_logger/formatters/data_dog.rb]: This can be `Rails.application.class.parent_name`. No need to pass it in as env var.

Comment (tavy88) [inline app/lib/semantic_logger/formatters/data_dog.rb]: You don't need to pass the app name as ENV:

You can use `Rails.application.class.parent_name`.

**Pull request merged**